### ISSUE TITLE
Fatal error when template file is empty

### ISSUE THREAD

**Issue by piercemoore** [cif/handlebar-rider#4]
**tl;dr:** This is awesome. I found a bug. I am fixing said bug. I will submit a pull request shortly.

The compiler fails when it hits an empty template file and leave the compiled template file output in an incomplete state.

```
[handlebar-rider] change detected to file, recompiling
[handlebar-rider] warning - deleted or now missing file 
[handlebar-rider] compiled template directory ./js/templates/ to ./js/templates.js
```

I have forked handlebar-rider and am currently writing the fix. I will submit a pull request just as soon as I get some detection and error handling built in. :+1: 

P.S. Handlebar-rider is fucking AWESOME. Thank you so much for taking the time to write this. Literally no other framework or module that I've found (and that is a fucking **lot**, btw) does namespaced template names or accommodates subfolders. Even a popular framework called Mimosa allows you to have subfolders, but you have to name all your template files uniquely, **regardless of where it is**. Which destroys the whole purpose.

**Comment by cif** [comment]
Glad you're stoked on it!  I've applied a namespaced approach to coffeescript compliation as well and use it in conjunction with .hb complication in the flint project.  

Good call on the empty file issue, always a pain if you're creating structure but not adding content and I always had to add files one at a time making sure to add placeholder before saving.  

Pull request away, I'll always merge.  Cheers!

**Comment by piercemoore** [comment]
1) Do you have any idea how cool it is to actually use a module that the developer is still interested in talking about? Very cool, not common.

2) I'm about to push up some commits to my fork of handlebar-rider, and I'm going to actually cause you some trouble by doing it (maybe). Here's why.

## Pull Request `1`

Actual bug fixes and some detection and some other small improvements, just as promised.

## Pull Request `2`,`3`,`4`.. (not sure)

Here's where it gets hairy, and I'd enjoy getting your feedback on this as well. I have been writing code on this for hours now and I have added a few things that I think would be awesome features and improve performance. Things like:

#### Underscore.js

> I'm a huge fanboy of Underscore to begin with, and it just makes things faster/easier. Not to mention that it makes the changes I made in this pull request much easier to implement.

#### Fully Asynchronous

> Yeah... The giant `for` loop that actually compiles all the templates could complete in a fraction of the time if made into an asynchronous function. So I'm going to give it a whirl and see how it goes. :-P

#### More default file extensions:

> `hb`,`hbs`,`handlebars`,`template`,`templates`,`tpl`,`item`,`part`,`prt`,`pt`

#### Ability to add more extensions in the command line, delimited by colons

> `handlebar-rider -i './path/to/templates/' -o './path/to/output.js' -e ext1:ext2:ext3

#### Formatted logging:

> I **promise** I am not trying to just inject my own code here. I wrote a lightweight logging library that colors and formats console output, and I have used it in a lot of projects.

#### Kinda sorta a refactor. But it's out of LOVE, I promise.

I work primarily in established projects at work that are not able to be converted into the structure required by most (read: all) of the projects that watch folders for changes. Because of this, I was just shit out of luck until I came across this awesome module. I cannot imagine that I am alone in wanting to use this and make my own development easier, so I am eager to contribute to make an already kick ass module even better. That's where this refactor comes in.

Interested to hear your thoughts!

**Comment by piercemoore** [comment]
Pull request #5 has been submitted.

**Comment by cif** [comment]
All that sounds awesome, pumped to have someone working on this as it is a tool I use everyday now.   

Seriously, **anything and everything** you can do to make it better under the hood is fully welcomed and I'll merge here and port into the Flint framework I'm creating.  I'm sick of modeling my data both on the client and server, Flint uses a common Flint.Model class and expressJS server to transact data.  Anyway, 

Big ups on async...  That for loop is gnarly.  I have a pretty big app I've been working on that has tons of individual template files and because it's not async I had to increase ulimits on my dev machines.  Hopefully your solution solves that problem

Have at 'er and pull request submit away, thanks!

**Comment by cif** [comment]
Oh regarding the console output formatter, let me know what you're using and I'll adopt that too.  Obviously mine is pretty basic/ghetto so I'm open to using a real one

**Comment by piercemoore** [comment]
Awesome!! I've been plotting changes and you should see some cool stuff coming up the pipe soon. I just submitted a pull request to fix a big mistake I made in package.json.

And the logging library I use is actually one I wrote myself, SuperCLI. https://github.com/piercemoore/SuperCLI

I'm excited to make this module all async and whatnot :-D
